pykafka/producer.py

```diff
@@ -434,17 +434,27 @@ def get_delivery_report(self, block=True, timeout=None):
 
     def _produce(self, message):
         """Enqueue a message for the relevant broker
-
+		Incase of  'success = False':
+			Retry to update metadata after self._max_retries.
+			Give up after self._max_retries * self._max_retries tries
         :param message: Message with valid `partition_id`, ready to be sent
         :type message: `pykafka.protocol.Message`
         """
         success = False
+        retry = 0
+        max_retries = self._max_retries ** 2
         while not success:
             leader_id = self._topic.partitions[message.partition_id].leader.id
             if leader_id in self._owned_brokers:
                 self._owned_brokers[leader_id].enqueue(message)
                 success = True
             else:
+                retry += 1
+                if retry % self._max_retries == 0:
+                    log.debug("Retries exceeded limit. Updating metadata again.")
+                    self._update()
+                elif retry > max_retries:
```

Comment:
This loop doesn't need to use `max_retries` at all. It's preferable to set the number of retries here to an arbitrary number - 10 should work.